src/paint.rs

```diff
@@ -138,18 +161,36 @@ impl<'a> Painter<'a> {
             Self::get_diff_style_sections(&self.minus_lines, &self.plus_lines, self.config);
 
         if self.config.side_by_side {
-            side_by_side::paint_minus_and_plus_lines_side_by_side(
+            let syntax_left_right = LeftRight::new(
                 minus_line_syntax_style_sections,
-                minus_line_diff_style_sections,
-                self.minus_lines.iter().map(|(_, state)| state).collect(),
                 plus_line_syntax_style_sections,
+            );
+            let diff_left_right = LeftRight::new(
+                minus_line_diff_style_sections,
                 plus_line_diff_style_sections,
+            );
```

Comment:
If "left" basically means the same thing as "minus" here, then the two sets of equivalent variable names here are making this code harder to understand. I'll wait for your opinion on this topic before merging this PR.